xray_test.go

```diff
@@ -950,3 +962,98 @@ func setStringFlags(flagSet *flag.FlagSet, flags ...string) []string {
 	}
 	return cmdFlags
 }
+
+// We perform validation on dependency resolution from an Artifactory server during the construction of the dependency tree. This process involves resolving all dependencies required by the project.
+func TestDependencyResolutionFromArtifactory(t *testing.T) {
+	initXrayTest(t, "")
+
+	testCases := []struct {
+		testProjectPath []string
+		resolveRepoName string
+		cacheRepoName   string
+		projectType     artUtils.ProjectType
+	}{
+		{
+			testProjectPath: []string{"npm", "npmproject"},
+			resolveRepoName: tests.NpmRemoteRepo,
+			cacheRepoName:   tests.NpmRemoteRepo,
+			projectType:     artUtils.Npm,
+		},
+		{
+			testProjectPath: []string{"nuget", "simple-dotnet"},
+			resolveRepoName: tests.NugetRemoteRepo,
+			cacheRepoName:   tests.NugetRemoteRepo,
+			projectType:     artUtils.Dotnet,
+		},
+		{
+			testProjectPath: []string{"yarn", "yarnproject"},
+			resolveRepoName: tests.YarnRemoteRepo,
+			cacheRepoName:   tests.YarnRemoteRepo,
+			projectType:     artUtils.Yarn,
+		},
+	}
+
+	for _, testcase := range testCases {
+		testSingleTechDependencyResolution(t, testcase.testProjectPath, testcase.resolveRepoName, testcase.cacheRepoName, testcase.projectType)
+	}
+}
+
+func testSingleTechDependencyResolution(t *testing.T, testProjectPartialPath []string, resolveRepoName string, cacheRepoName string, projectType artUtils.ProjectType) {
+	log.Info(fmt.Sprintf("--- Starting %s test ---", projectType.String()))
+	tempDirPath, createTempDirCallback := coretests.CreateTempDirWithCallbackAndAssert(t)
+	defer createTempDirCallback()
+	testProjectPath := filepath.Join(append([]string{filepath.FromSlash(tests.GetTestResourcesPath())}, testProjectPartialPath...)...)
+	assert.NoError(t, biutils.CopyDir(testProjectPath, tempDirPath, true, nil))
+	rootDir, err := os.Getwd()
+	require.NoError(t, err)
+	require.NoError(t, os.Chdir(tempDirPath))
+	defer func() {
+		assert.NoError(t, os.Chdir(rootDir))
+	}()
+
+	cleanUpJfrogHome, err := coretests.SetJfrogHome()
+	assert.NoError(t, err)
+	defer cleanUpJfrogHome()
+	server := &config.ServerDetails{
+		Url:            *tests.JfrogUrl,
+		ArtifactoryUrl: *tests.JfrogUrl + tests.ArtifactoryEndpoint,
+		AccessToken:    *tests.JfrogAccessToken,
+		ServerId:       "test",
+	}
+
+	configCmd := coreCmd.NewConfigCommand(coreCmd.AddOrEdit, "test").SetDetails(serverDetails).SetUseBasicAuthOnly(true).SetInteractive(false)
+	assert.NoError(t, configCmd.Run())
+	context := createContext(t, "repo-resolve="+resolveRepoName, "server-id-resolve="+serverDetails.ServerId, "global=false")
```

Comment:
```suggestion
	context := createContext(t, "repo-resolve="+resolveRepoName)
```
Explanation:
1. Use the default server ID
2. Remove global=false because it is the default